examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,650 @@
+    { 9, 2452 },   { 10, 2457 },  { 11, 2462 },  { 12, 2467 },  { 13, 2472 },  { 14, 2484 },  { 32, 5160 },  { 34, 5170 },
+    { 36, 5180 },  { 38, 5190 },  { 40, 5200 },  { 42, 5210 },  { 44, 5220 },  { 46, 5230 },  { 48, 5240 },  { 50, 5250 },
+    { 52, 5260 },  { 54, 5270 },  { 56, 5280 },  { 58, 5290 },  { 60, 5300 },  { 62, 5310 },  { 64, 5320 },  { 68, 5340 },
+    { 96, 5480 },  { 100, 5500 }, { 102, 5510 }, { 104, 5520 }, { 106, 5530 }, { 108, 5540 }, { 110, 5550 }, { 112, 5560 },
+    { 114, 5570 }, { 116, 5580 }, { 118, 5590 }, { 120, 5600 }, { 122, 5610 }, { 124, 5620 }, { 126, 5630 }, { 128, 5640 },
+    { 132, 5660 }, { 134, 5670 }, { 136, 5680 }, { 138, 5690 }, { 140, 5700 }, { 142, 5710 }, { 144, 5720 }, { 149, 5745 },
+    { 151, 5755 }, { 153, 5765 }, { 155, 5775 }, { 157, 5785 }, { 159, 5795 }, { 161, 5805 }, { 165, 5825 }, { 169, 5845 },
+    { 173, 5865 }, { 183, 4915 }, { 184, 4920 }, { 185, 4925 }, { 187, 4935 }, { 188, 4940 }, { 189, 4945 }, { 192, 4960 },
+    { 196, 4980 },
+} });
+
+constexpr size_t kUdpBufferSize = 512;
+
+constexpr uint8_t kWiFiConnectRetryMax = 5;
+
+/* The event group allows multiple bits for each event, but we only care about
+ * whether the STA initialization was successful
+ */
+constexpr uint8_t WIFI_STA_UP_BIT = BIT0;
+
+/* The event group allows multiple bits for each event, but we only care about 3 events:
+ * - we are connected to the AP with an IP6
+ * - we are connected to the AP with an IP4
+ * - we failed to connect after the maximum amount of retries */
+#define WIFI_CONNECTED_IP4_BIT BIT1
+#define WIFI_CONNECTED_IP6_BIT BIT2
+#define WIFI_FAIL_BIT BIT3
+
+SemaphoreHandle_t wifi_init_semaphore;
+SemaphoreHandle_t scanresults_semaphore;
+
+/* FreeRTOS event group to signal when we are connected*/
+EventGroupHandle_t s_wifi_event_group;
+
+int s_retry_num        = 0;
+esp_netif_t * wifi_sta = nullptr;
+
+int sockfd = 0;
+unsigned int clientlen;         // byte size of client's address
+struct sockaddr_in6 clientaddr; // client addr
+
+::pw::rpc::ServerWriter<chip_rpc_ScanResults> scanresult_writer;
+
+class RAIIGuard
+{
+    std::function<void()> release_;
+
+public:
+    RAIIGuard(std::function<void()> acquire, std::function<void()> release) : release_(release) { acquire(); }
+    virtual ~RAIIGuard() { release_(); }
+};
+
+void wifi_start_event_handler(void * arg, esp_event_base_t event_base, int32_t event_id, void * event_data)
+{
+    if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_START)
+    {
+        ESP_LOGI(TAG, "EVENT: WIFI_EVENT_STA_START");
+        xEventGroupSetBits(s_wifi_event_group, WIFI_STA_UP_BIT);
+    }
+}
+
+void wifi_conn_event_handler(void * arg, esp_event_base_t event_base, int32_t event_id, void * event_data)
+{
+    if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_DISCONNECTED)
+    {
+        ESP_LOGI(TAG, "EVENT: WIFI_EVENT_STA_DISCONNECTED, reason: %d",
```

Comment:
Can this use generic CHIP stack loggingÉ

Reply:
It could, but most of the other examples have used the style of using the platform logging in the application code.